### PR TITLE
Data: Always use delete index for position deletes

### PR DESCRIPTION
**Pull request by aokolnychyi**
This PR removes the outdated logic in `DeleteFilter` that determined whether we have to stream through position deletes or load them in memory. Instead, we should always rely on the bitmap-based approach like in vectorized reads.

### CODE REVIEW

File: data/src/main/java/org/apache/iceberg/data/DeleteFilter.java [hunk 1]
```diff
@@ -245,18 +242,9 @@ private CloseableIterable<T> applyPosDeletes(CloseableIterable<T> records) {
 
     List<CloseableIterable<Record>> deletes = Lists.transform(posDeletes, this::openPosDeletes);
 
-    // if there are fewer deletes than a reasonable number to keep in memory, use a set
```
Comment (aokolnychyi) [inline data/src/main/java/org/apache/iceberg/data/DeleteFilter.java:248]: When this logic was added a few years ago, we added position deletes into a set. We have been using bitmaps for a while now. In fact, vectorized reads always build bitmaps and have no threshold on the number of deletes. This has proven to work really well. Position deletes represented as bitmaps should always fit in memory.

Position deletes compress really well both on disk and in memory. We have seen this 100K threshold causing degradation in jobs without any good reason.

**Comment by aokolnychyi** [comment]
cc @amogh-jahagirdar @nastra @RussellSpitzer @flyrain @Fokko

File: data/src/main/java/org/apache/iceberg/data/DeleteFilter.java (hunk 1 above)
Comment (amogh-jahagirdar) [reply data/src/main/java/org/apache/iceberg/data/DeleteFilter.java:248]: Sweet, this makes sense and appreciate the simplification.  I remember seeing this threshold in the code and was wondering how we got to the 100000 value.

**Review by amogh-jahagirdar — APPROVED** [review]

**Review by nastra — APPROVED** [review]
LGTM, thanks @aokolnychyi for improving this

**Review by singhpk234 — APPROVED** [review]
LGTM, Thanks @aokolnychyi !

**Comment by aokolnychyi** [comment]
Thank you for reviewing, @amogh-jahagirdar @nastra @singhpk234!

**Pull request merged**